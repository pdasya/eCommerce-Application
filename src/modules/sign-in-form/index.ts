export { SignInForm } from './sign-in-form-module/sign-in-form.module';
