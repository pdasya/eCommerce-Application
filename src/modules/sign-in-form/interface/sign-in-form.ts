export interface IUserDraft {
  email: string;
  password: string;
  [key: string]: string;
}
