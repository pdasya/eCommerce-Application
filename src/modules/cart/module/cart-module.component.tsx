import { Button, Grid, Typography } from '@mui/material';
import { Box } from '@mui/system';
import React, { FC, useEffect, useState } from 'react';
import { toast } from 'react-toastify';
import { selectCart } from '@store/cart/cart.slice';
import { useAppSelector } from '@hooks/use-app-selector.hook';
import { cartService } from '@modules/cart/services/cart.service';
import { selectAuthorization } from '@store/auth/auth.slice';
import { CartHeader } from '../components/cart-header/cart-header';
import { CartItemList } from '../components/cart-item-list/cart-item-list';
import { PromoCodeForm } from '../components/cart-promocode-form/cart-promocode-form';
import { CartActions } from '../components/cart-actions/cart-actions';
import { ICartItem } from '../interfaces/cart-item.interface';
import * as styles from './cart-module.component.module.scss';

export const CartModule: FC = () => {
  const cart = useAppSelector(selectCart);
  const isAuthorized = useAppSelector(selectAuthorization);

  const [loading, setLoading] = useState(false);
  const [promoCode, setPromoCode] = useState('');
  const [promoCodeId, setPromoCodeId] = useState('');
  const [promoCodeState, setPromoCodeState] = useState(false);
  const [promoError, setPromoError] = useState('');

  const getPromoName = async () => {
    try {
      const results = await cartService.getPromoCodeInfo();
      if (results) {
        if (results.promoName) {
          setPromoCode(results.promoName);
          setPromoCodeId(results.ID);
          setPromoCodeState(true);
        }
      }
    } catch (error) {
      toast.error(error);
    }
  };

  useEffect(() => {
    getPromoName();
  }, [promoCodeState]);

  useEffect(() => {
    setLoading(true);
    cartService
      .synchronize()
      .catch(error => toast.error(error))
      .finally(() => setLoading(false));
  }, [isAuthorized]);

  const addMockItems = async () => {
    setLoading(true);

    const items = [
      {
        sku: 'yamato-sauces-3_1',
        quantity: 1,
      },
      {
        sku: 'itoen-tea-1_1',
        quantity: 1,
      },
      {
        sku: 'itoen-tea-4_1',
        quantity: 1,
      },
      {
        sku: 'itoen-tea-3_1',
        quantity: 1,
      },
    ];

    try {
      await cartService.addCartItem(items);
      setLoading(false);
    } catch (error) {
      toast.error('Error fetching cart:', error);
    } finally {
      setLoading(false);
    }
  };

  const promoCodeRemover = async () => {
    if (promoCode) {
      await cartService.removePromoCode(promoCodeId);
    }
    setPromoCode('');
    setPromoCodeState(false);
  };

  const removeItemHandler = async (item: ICartItem) => {
    try {
      await cartService.removeCartItem(item);
      toast.success(`Item ${item.name} is successfully deleted from the cart`);
    } catch (error) {
      toast.error(`Error deleting item`);
    }
  };

  const incrementItemHandler = async (item: ICartItem) => {
    try {
      cartService.updateCartItemQuantity([{ ...item, quantity: item.quantity + 1 }]);
      toast.success(`Item ${item.name} quantity successfully changed`);
    } catch (error) {
      toast.error(`Error updating quantity: ${error}`);
    }
  };

  const decrementItemHandler = async (item: ICartItem) => {
    try {
      cartService.updateCartItemQuantity([{ ...item, quantity: item.quantity + -1 }]);
      toast.success(`Item ${item.name} quantity successfully changed`);
    } catch (error) {
      toast.error(`Error updating quantity: ${error}`);
    }
  };

  const handleCartClearItems = async () => {
    if (window.confirm('Are you sure you want to clear the cart?')) {
      setLoading(true);
      try {
        await promoCodeRemover();
        await cartService.deleteCart();
        toast.success('Cart has been cleared');
      } catch (error) {
        toast.error('Error clearing cart');
      } finally {
        setLoading(false);
      }
    }
  };

  const handlePromoCodeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    setPromoCode(value);
    setPromoError('');
  };

  const handlePromoCodeApplyItems = async () => {
    try {
      await cartService.applyPromoCode(promoCode);
      toast.success(`Promo code "${promoCode}" is successfully applied`);
      setPromoCodeState(true);
    } catch (error) {
      setPromoError('Invalid promo code');
      toast.warning(`Unable apply promo code: ${error.message}`);
    }
  };

  const handlePromoCodeDeleteItems = async () => {
    try {
      await promoCodeRemover();
      toast.success(`Promo code "${promoCode}" is successfully deleted`);
    } catch (error) {
      setPromoError('Invalid promo code');
      toast.warning(`Unable deleted promo code: ${error.message}`);
    }
  };

  const handlePromoCodePressEnter = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter' && promoCode.length > 0) {
      handlePromoCodeApplyItems();
    }
  };

  return (
    <Grid container spacing={2} className={styles.cartModuleWrapper}>
      <CartHeader />
      <Grid item xs={12}>
        <Box className={styles.cartModuleContent}>
          {loading ? (
            <Typography variant="body1" gutterBottom>
              Loading...
            </Typography>
          ) : cart.items.length === 0 ? (
            <Typography variant="body1" gutterBottom>
              Your cart is empty
            </Typography>
          ) : (
            <>
              <CartItemList
                cartItems={cart.items}
                onIncrement={incrementItemHandler}
                onDecrement={decrementItemHandler}
                onRemove={removeItemHandler}
              />
              <Box textAlign="right" marginTop={2}>
                {cart.initialPrice !== cart.finalPrice && (
                  <Typography variant="body2" style={{ textDecoration: 'line-through' }}>
                    ${cart.initialPrice.toFixed(2)}
                  </Typography>
                )}
                <Typography variant="h6">Total Price: ${cart.finalPrice.toFixed(2)}</Typography>
              </Box>
              <PromoCodeForm
                promoCode={promoCode}
                promoCodeState={promoCodeState}
                promoError={promoError}
                handlePromoCodeChange={handlePromoCodeChange}
                handlePromoCodePressEnter={handlePromoCodePressEnter}
                handlePromoCodeApply={handlePromoCodeApplyItems}
                handlePromoCodeDeleteItems={handlePromoCodeDeleteItems}
              />
              <CartActions handleCartClear={handleCartClearItems} />
            </>
          )}
          {cart.items.length === 0 && (
            <Button variant="contained" color="info" onClick={addMockItems}>
              Add Cart Items
            </Button>
          )}
        </Box>
      </Grid>
    </Grid>
  );
};
