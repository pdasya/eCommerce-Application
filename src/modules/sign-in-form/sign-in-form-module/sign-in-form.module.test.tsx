import React from 'react';
import { fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import { renderWithProviders } from '@utils/render-with-providers.test-util';
import { SignInForm } from './sign-in-form.module';

describe('SignIn', () => {
  test('email input accept text', async () => {
    const screen = renderWithProviders(<SignInForm />);
    const form = screen.container.querySelector('form') as HTMLFormElement;
    const emailInputNode = screen.container.querySelector('input[name=email]') as HTMLInputElement;

    expect(emailInputNode.value).toMatch('');
    await act(() => fireEvent.change(emailInputNode, { target: { value: 'testing' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('testing');

    let errorMessage = screen.getByText('Email must be at example user@example.com');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: '@test.com' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('@test.com');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: 'test@test' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('test@test');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: ' test@test.com' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch(' test@test.com');
    errorMessage = screen.getByText('Email should not contain spaces');

    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: 'test@test.com ' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('test@test.com ');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: 'test @test.com' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('test @test.com');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(emailInputNode, { target: { value: 'test@test.com' } }));
    await act(() => fireEvent.submit(form));
    expect(emailInputNode.value).toMatch('test@test.com');
    expect(errorMessage).not.toBeInTheDocument();
  });

  test('snapshot match', () => {
    const { asFragment } = renderWithProviders(<SignInForm />);
    expect(asFragment()).toMatchSnapshot('sign-in-form-snapshot');
  });

  test('password input accept text', async () => {
    const screen = renderWithProviders(<SignInForm />);
    const form = screen.container.querySelector('form') as HTMLFormElement;
    const passwordInputNode = screen.container.querySelector(
      'input[name=password]',
    ) as HTMLInputElement;

    expect(passwordInputNode.value).toMatch('');
    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'testing' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('testing');

    let errorMessage = screen.getByText('Password must be at least 8 characters long');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'passworld' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('passworld');

    errorMessage = screen.getByText(
      'Password must contain at least one uppercase letter, one lowercase letter, and one number',
    );
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'pass1234' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('pass1234');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'passWORLD' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('passWORLD');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: '12345678' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('12345678');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: ' paSS1234' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch(' paSS1234');
    expect(errorMessage).toBeInTheDocument();

    errorMessage = screen.getByText('Password should not contain spaces');

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'paSS1234 ' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('paSS1234 ');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'paSS 1234' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('paSS 1234');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'pass1234' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('pass1234');
    expect(errorMessage).toBeInTheDocument();

    await act(() => fireEvent.change(passwordInputNode, { target: { value: 'paSS1234' } }));
    await act(() => fireEvent.submit(form));
    expect(passwordInputNode.value).toMatch('paSS1234');

    expect(errorMessage).not.toBeInTheDocument();
  });

  test('snapshot match', () => {
    const { asFragment } = renderWithProviders(<SignInForm />);
    expect(asFragment()).toMatchSnapshot('sign-in-form-snapshot');
  });
});
