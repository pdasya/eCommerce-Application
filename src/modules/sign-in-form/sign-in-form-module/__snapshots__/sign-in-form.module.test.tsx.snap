// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn snapshot match: sign-in-form-snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 form css-3su884-MuiPaper-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-6exuqi-MuiTypography-root"
    >
      Sign In
    </h2>
    <form
      action="#"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1eed5fa-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="on"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r2:"
                  name="email"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth passwordInput css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                form="[object Object]"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r3:"
                  id=":r3:-label"
                >
                  Password
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1eed5fa-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="on"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id=":r3:"
                    name="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                class="passwordIcon"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="link registerLink"
        href="/sign-up"
      >
        Don't have an account yet? Register
      </a>
    </form>
  </div>
</DocumentFragment>
`;

exports[`SignIn snapshot match: sign-in-form-snapshot 2`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 form css-3su884-MuiPaper-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-6exuqi-MuiTypography-root"
    >
      Sign In
    </h2>
    <form
      action="#"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r6:"
                id=":r6:-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1eed5fa-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="on"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r6:"
                  name="email"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth passwordInput css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                form="[object Object]"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r7:"
                  id=":r7:-label"
                >
                  Password
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1eed5fa-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="on"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id=":r7:"
                    name="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                class="passwordIcon"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Login
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        class="link registerLink"
        href="/sign-up"
      >
        Don't have an account yet? Register
      </a>
    </form>
  </div>
</DocumentFragment>
`;
